internal/selfmonitor/config/config_builder.go

```diff
@@ -38,11 +38,58 @@ func makeScrapeConfig(scrapeNamespace string) []ScrapeConfig {
 		{
 			JobName: "kubernetes-service-endpoints",
 
-			RelabelConfigs: []RelabelConfig{{
-				SourceLabels: []string{"__meta_kubernetes_service_annotation_prometheus_io_scrape"},
-				Regex:        "true",
-				Action:       Keep,
-			}},
+			RelabelConfigs: []RelabelConfig{
+				{
+					SourceLabels: []string{"__meta_kubernetes_namespace"},
+					Action:       Keep,
+					Regex:        scrapeNamespace,
+				},
+				{
+					SourceLabels: []string{"__meta_kubernetes_service_annotation_prometheus_io_scrape"},
+					Action:       Keep,
+					Regex:        "true",
+				},
+				{
+					SourceLabels: []string{"__meta_kubernetes_service_label_app_kubernetes_io_name"},
+					Action:       Keep,
+					Regex:        "telemetry-.+",
+				},
```

Comment:
As discussed, instead of relying on app.kubernetes.io/name label, let's label the OTel Collectors with a special marker label and use it in the scrape config filter.

Reply:
done